packages/fuel-indexer-macros/src/indexer.rs

```diff
@@ -537,16 +537,22 @@ fn process_fn_items(
 
                     let mut return_types = Vec::new();
 
+                    // Track callee contract IDs from Call receipts
+                    let mut callees = HashSet::new();
+
                     for receipt in tx.receipts {
                         match receipt {
-                            Receipt::Call { param1, id, ..} => {
+                            Receipt::Call { param1, id, to, ..} => {
                                 #contract_conditional
                                 return_types.push(param1);
+                                callees.insert(to);
                             }
                             Receipt::ReturnData { data, id, .. } => {
                                 #contract_conditional
-                                let selector = return_types.pop().expect("No return type available. <('-'<)");
-                                decoder.decode_return_type(selector, data);
+                                if callees.contains(&id) {
```

Comment:
@deekerno  
- I think this `callee` logic is fair. 
- Looking over the issue https://github.com/FuelLabs/fuel-indexer/issues/449 it looks like the `callee` logic works also with `Receipt::Return` in addition to `Receipt::ReturnData`
- Meaning, if we add `Receipt::Return` and have it look in the `callee` as well, I think this still fixes the bug, as well as addresses the case where you have a `Receipt::Return` and a `Receipt::ReturnData`, but only the `Receipt::Return` having a valid `id` param

Here's the example from the issue:
```
receipts: [
  Call { id: 0000000000000000000000000000000000000000000000000000000000000000, to: 69ec8b0544b6549354e373e021fc874305176c7b81db6da8d16cded3950d9f9f, amount: 0, asset_id: 0000000000000000000000000000000000000000000000000000000000000000, gas: 99970291, param1: 4000556463, param2: 0, pc: 17216, is: 17216 },
  LogData { id: 69ec8b0544b6549354e373e021fc874305176c7b81db6da8d16cded3950d9f9f, ra: 0, rb: 6846832, ptr: 67108626, len: 77, digest: ccbcf7ee5796d9316fd3472b561feb99a51d3d97e82426b75df4a093faedc890, data: [0, 0, 0, 0, 0, 102, 117, 101, 108, 107, 99, 128, 76, 251, 249, 133, 110, 104, 229, 182, 231, 174, 242, 56, 220, 131, 17, 236, 85, 190, 192, 77, 247, 116, 0, 58, 44, 150, 224, 65, 142, 0, 0, 0, 4, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], pc: 25284, is: 17216 },
  Return { id: 69ec8b0544b6549354e373e021fc874305176c7b81db6da8d16cded3950d9f9f, val: 69, pc: 25292, is: 17216 },
  ReturnData { id: 0000000000000000000000000000000000000000000000000000000000000000, ptr: 67108544, len: 160, digest: de5c8876a60211bd60e33fed909e43cc8762a364833eb5c0014a745283045d78, data: [0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 69, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 48, 184, 0, 0, 0, 0, 0, 0, 58, 152, 0, 0, 0, 0, 0, 0, 64, 232, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 3, 255, 255, 95, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 5, 245, 98, 243, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 40, 104, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 5, 245, 108, 243, 0, 0, 0, 0, 0, 0, 59, 240], pc: 11496, is: 10344 },
  ScriptResult { result: Success, gas_used: 135105 }], id: 371821733e4c23b30791fb078dd269b58e2481a3ec662a13ee1fc047fce62c5c }
]
```
- With your existing code, we would have pushed a `Receipt::Call`, but we wouldn't have popped anything because the `Receipt::ReturnData`'s `id` is invalid, but the `Receipt::Return`'s `id` _is_ valid

Reply:
That sounds good; I'm working on `Return` receipt support right now, so I'll ping you for a re-review once that's merged and this is adjusted.

Reply:
+1 for same PR